### DISCUSSION TITLE
Database indexes

### DISCUSSION THREAD

**Discussion by amos-kibet** [3Stones-io/omedis#389] [Q&A] [answered]
Is it necessary to create indexes for foreign keys? I see that we don't add them, and was wondering if it is intentional or not. Do query JOINs and loads/preloads benefit from such indexes?

cc @almirsarajcic, @wintermeyer.

**Comment by wintermeyer** [comment, marked as answer]
Yes, good catch @amos-kibet !

General Rule about index: When in doubt, add it. Why? The index only slows down the write but not the read. So the worst case scenario is that we loose a millisecond in the write of a new data set and never use the index.

One more thing about index: If you have an index `[a]` and an index `[a, b]` you can delete the first one. A typical SQL database can use the `[a, b]` index to search for just `a` too and by just having one index to load it is easier to cache it and to increase the speed.

**Comment by amos-kibet** [comment]
Thanks @wintermeyer for the elaborate explanation.
